### PR TITLE
Added peak tagging plugin

### PR DESCRIPTION
**Pull request by WenzDaniel**
**What is the problem / what does the code in this PR do**
In this PR we add the Peak counterpart for the veto regions plugins which are added in https://github.com/XENONnT/straxen/pull/465. The working principle is simple. We tag all S1 peaks which touch one of the veto regions. The tag is cumulative in case both veto systems issued a veto signal. In addition we are computing the time difference to the closest veto interval.

- [x] Merge https://github.com/XENONnT/straxen/pull/465
- [x] Add a new event level plugin above event_info.
- [ ] Add tests.

**Proposed Dependency structure:**
To minimize the interference with TPC processing two new data_types are introduced. One at the peak level and one above event_info. They are shown in the subsequent figure. 

In this way we can process the TPC up to event_info without the need of having the data of the vetos already processed. Both data types are only save when asked, e.g. for dedicated studies about the optimization of the veto interval length. Computational wise are both plugins very light weighted.  

DependencyVetoTagging

### CODE REVIEW

File: straxen/plugins/peak_processing.py [hunk 1]
```diff
@@ -324,3 +324,85 @@ def find_n_competing(peaks, windows, fraction):
...
+    """
+    Plugin which tags S1 peaks according to  muon and neutron-vetos.
+    Tagging S2s is does not make sense as they occur with a delay.
+    However, we compute for both S1/S2 the time delay to the closest veto
+    region.
+
+        * untagged: 0
+        * neutron-veto: 1
+        * muon-veto: 2
+        * both vetos: 3
```
Comment (jorana) [inline straxen/plugins/peak_processing.py:339]: Would it make sense to make this an int-num like https://github.com/AxFoundation/strax/blob/master/strax/processing/data_reduction.py#L12? That would make it easier not to accidentally mix up the numbers?

**Review by jorana — COMMENTED** [review]
I thought you requested my review but it appears to be still a draft. Here some minor comments/suggestions but I'll re-review after it's undrafted

File: straxen/plugins/peak_processing.py [hunk 2]
```diff
@@ -324,3 +324,85 @@ def find_n_competing(peaks, windows, fraction):
...
+    return tags
+
+
+def get_time_to_closest_veto(peaks, veto_intervals):
+    """
+    Computes time difference between peak and closest veto interval.
+    """
+    vetos = np.zeros(len(veto_intervals)+2, np.int64)
+    vetos[1:-1] = veto_intervals['time']
+    vetos[-1] = 9223372036854775807  # 64 bit infinity
```
Comment (jorana) [inline straxen/plugins/peak_processing.py:387]: perhaps we could re-use https://github.com/XENONnT/straxen/blob/master/straxen/plugins/acqmon_processing.py#L10

File: straxen/plugins/event_info.py [hunk 3]
```diff
@@ -1,57 +1,113 @@
...
+    This only uses 1T data-types as several event-plugins are nT only
+    """
+    depends_on = ['events',
+                  'event_basics',
+                  'event_positions',
+                  'corrected_areas',
+                  'energy_estimates']
+    provides = 'event_info'
+    save_when = strax.SaveWhen.ALWAYS
+    __version__ = '0.0.0'
```
[the remark this answers is not in the document]
Comment (WenzDaniel) [reply straxen/plugins/event_info.py:60]: Ups yes, for sure.

File: straxen/plugins/event_info.py [hunk 4]
```diff
@@ -1,57 +1,113 @@
...
+        split_tags = strax.split_by_containment(peaks, events)
+        result = np.zeros(len(events), self.dtype)
+        result['time'] = events['time']
+        result['endtime'] = events['endtime']
+        get_veto_tags(events, split_tags, result)
+
+        return result
+
+
+def get_veto_tags(events, split_tags, result):
```
[the remark this answers is not in the document]
Comment (WenzDaniel) [reply straxen/plugins/event_info.py:97]: Only if field names are hardcoded as f-strings are not working yet in numba. But I can do.

[the remark this answers is not in the document]
Comment (jorana) [reply straxen/plugins/event_info.py:97]: ah yeah, you are right, the biggest omission in numba functionality 😭

File: straxen/plugins/peak_processing.py (hunk 2 above)
Comment (WenzDaniel) [reply straxen/plugins/peak_processing.py:387]: Sure, but I would put also "infinity" there. @AlexElykov do you see any problem if we replace the 1h in `T_NO_VETO_FOUND` by a 64 bit infinity? 1 h is a bit arbitrary.

File: straxen/plugins/event_info.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (WenzDaniel) [reply straxen/plugins/event_info.py:97]: Just for the record. I just checked with a 1h run it takes 
```
643 ms ± 20.2 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```
to compute `get_veto_tags`. So no need for numba.

File: straxen/plugins/event_info.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (WenzDaniel) [reply straxen/plugins/event_info.py:60]: Hmm somehow it is not revertible.  I guess I will have to make a new branch....

**Comment by WenzDaniel** [comment]
Somehow it is not possible to clean the git history. I made a new branch.

**Pull request closed without merging**